ts/src/index.ts

```diff
@@ -129,14 +130,36 @@ export async function start(config: Config = {}): Promise<void> {
         config);
     return;
   }
+
+  // start the heap profiler if profiler config does not indicate heap profiling
+  // is disabled. This must be done before any asynchronous calls are made so
+  // samples from the first tick can be captured.
+  let heapProfiler: HeapProfiler|undefined;
+  if (!config.disableHeap) {
+    heapProfiler = new HeapProfiler(
```

Comment:
Creating the heap profiler seems overkill. Can you have enable() / disable() methods as static on the heap profiler class (or just as functions) instead and call those?

Reply:
Also make enable() / disable() set and check the global boolean flag for whether the heap profiler is enabled. Then you can call disable() without a check around it.

Reply:
I removed the heap profiler class. I think doing this addresses this comments, an also makes sense because (unlike the time profiler) the heap profiler can only be collecting a profile with one set of parameters.